### PR TITLE
GfHEPKRG: Handle Single IDP journey crashing when transaction entity …

### PR DESCRIPTION
**Pull request by JohnWattsGDS**
…id missing

Enable validation for missing transaction entity id.

Co-authored-by: Benjamin Gill <benjamin.gill@digital.cabinet-office.gov.uk>

### CODE REVIEW

File: app/models/session_validator/transaction_entity_id_presence.rb [hunk 1]
```diff
@@ -3,14 +3,12 @@ class TransactionEntityIdPresence
     ERROR_MESSAGE = "Transaction entity ID can not be found in the user's session".freeze
     def validate(_cookies, session)
       # When confident of no false positives, remove the log and reinstate the validation failure.
```
Comment (bjgill) [inline app/models/session_validator/transaction_entity_id_presence.rb]:
```suggestion
```

**Comment by bjgill** [comment]
Also, is the commit message/PR title accurate?

**Comment by JohnWattsGDS** [comment]
Yes, I believe the title is fine.  That's what we're doing; we're handling the case where the transaction entity id is missing.  Originally it was left to cause problems later, then we logged that it had happened but did nothing else and now we handle it in the only way we can - by halting processing and letting the user know that something went wrong.

File: app/models/session_validator/transaction_entity_id_presence.rb [hunk 2]
```diff
@@ -3,14 +3,11 @@ class TransactionEntityIdPresence
...
```
Comment (bjgill) [inline app/models/session_validator/transaction_entity_id_presence.rb]: Are you sure we want to keep this comment?

Comment (JohnWattsGDS) [reply app/models/session_validator/transaction_entity_id_presence.rb]: Didn't even notice it.

**Comment by galund** [comment]
You should still fix Benjamin's email address in the commit message :-)

**Comment by JohnWattsGDS** [comment]
Indeed I should - I thought I'd got away with editing it above but I've done it for real now.

**Pull request merged**